### ISSUE TITLE
RED cannot run a folder suite with "." in name

### ISSUE THREAD

**Issue by yangxj1339** [nokia/RED#262]
1. New a  project and a folder named like "Folder.1".
2. New a robot file in Folder.1 and create a testcase;
3. Run on Folder.1, then RED will report an error: contains no test in suite "Folder";

This is occoured in /org.robotframework.ide.main.plugin/src/org/robotframework/ide/eclispe/main/plugin/launch/local/LocalProcessCommandLineBuilder.addDataSource. When convert a resource to path segments,  A "removeFileExtension" method is called  either the resource is a file or a folder.

**Comment by michalanglart** [comment]
Hi,

it is not related with `LocalProcessCommandLineBuilder#addDataSource` method and moreover it is a bug rather in Robot than in RED.

Given following structure e.g.:
```
X
|___ Y
     |___ Z.robot
```

there are two possibilities to run only tests contained within `Y` and both can be used by RED depending if `Preferences -> Robot Framework -> Launching -> Pass selected suite as data source` preference is checked (option (2)) or not (option (1)):

1. Run robot by passing path to `X` as data source and use `--suite` selector to only select `Y`:

```
python -m robot.run --suite X.Y d:\path\to\X
```

2. Run robot by passing path to `Y` as data source

```
python -m robot.run d:\path\to\X\Y
```

The problem occurs with option (1) (which is default in RED) when `Y` contains dots inside it's name. The problems comes from the fact that `.` character is used in `--suite` selector to separate nested suites. Same applies to `--test` selector used to select tests inside suites.

However if you will try to run option (1) from command line you will observe the same result and Robot will not execute any tests. 

From what I've found there was similar issue raised against RF in past but only for case of files (the last segment in `--suite` selector): https://github.com/robotframework/robotframework/issues/1632

From our point of view either option (2) should be used (but it also have some implications: for example tree in `Execution` view will look differently for both options) or this should be fixed on RF side as there is no possibility to escape dots used in names when passing them through `--suite` selector.

**Comment by michalanglart** [comment]
Hi,

It seems that both you and I were right ;)

Yesterday I haven't checked that in RED actually and I looked into Robot with command line straight away. However you're right because we do cut off the prefix after `.` character.

On the other hand I'm also right because it does not work in Robot - but in version 3.1 beta 1 which I'm currently using. The same `--suite` selector with dots in folder names works in Robot 3.1 alpha 2.

We'll fix that on our side as well as I'm going to raise an issue on RF github.

**Comment by michalanglart** [comment]
Fixed and released in 0.8.10
